backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/types/dsl/impl/ElasticsearchStringIndexFieldTypeOptionsStep.java

```diff
@@ -202,6 +203,7 @@ public IndexFieldType<String> toIndexFieldType() {
 		if ( resolvedProjectable ) {
 			builder.projectable( true );
 			builder.queryElementFactory( ProjectionTypeKeys.FIELD, new ElasticsearchFieldProjection.Factory<>( codec ) );
+			builder.queryElementFactory( ProjectionTypeKeys.HIGHLIGHT, new ElasticsearchFieldHighlightProjection.Factory<>() );
```

Comment:
Doesn't this only make sense for analyzed (full-text) fields? I'm not sure how highlighting would work on keyword fields...